python/tests/catalog/test_hive.py

```diff
@@ -0,0 +1,386 @@
+            outputFormat="org.apache.hadoop.mapred.FileOutputFormat",
+            compressed=False,
+            numBuckets=0,
+            serdeInfo=SerDeInfo(
+                name=None,
+                serializationLib="org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe",
+                parameters={},
+                description=None,
+                serializerClass=None,
+                deserializerClass=None,
+                serdeType=None,
+            ),
+            bucketCols=[],
+            sortCols=[],
+            parameters={},
+            skewedInfo=SkewedInfo(skewedColNames=[], skewedColValues=[], skewedColValueLocationMaps={}),
+            storedAsSubDirectories=False,
+        ),
+        partitionKeys=[],
+        parameters={"EXTERNAL": "TRUE", "transient_lastDdlTime": "1659092339"},
+        viewOriginalText=None,
+        viewExpandedText=None,
+        tableType="EXTERNAL_TABLE",
+        privileges=None,
+        temporary=False,
+        rewriteEnabled=False,
+        creationMetadata=None,
+        catName="hive",
+        ownerType=1,
+        writeId=-1,
+        isStatsCompliant=None,
+        colStats=None,
+        accessType=None,
+        requiredReadCapabilities=None,
+        requiredWriteCapabilities=None,
+        id=None,
+        fileMetadata=None,
+        dictionary=None,
+        txnId=None,
+    )
+
+
+@pytest.fixture
+def hive_database() -> HiveDatabase:
+    return HiveDatabase(
+        name="default",
+        description=None,
+        locationUri="file:/tmp/default2.db",
+        parameters={"test": "property"},
+        privileges=None,
+        ownerName=None,
+        ownerType=1,
+        catalogName="hive",
+        createTime=None,
+        managedLocationUri=None,
+        type=None,
+        connector_name=None,
+        remote_dbname=None,
+    )
+
+
+def test_check_number_of_namespaces(table_schema_simple: Schema):
+    catalog = HiveCatalog(HIVE_CATALOG_NAME, {}, url=HIVE_METASTORE_FAKE_URL)
+
+    with pytest.raises(ValueError):
+        catalog.create_table(("default", "namespace", "table"), schema=table_schema_simple)
+
+    with pytest.raises(ValueError):
+        catalog.create_table("default.namespace.table", schema=table_schema_simple)
+
+    with pytest.raises(ValueError):
+        catalog.create_table(("table",), schema=table_schema_simple)
+
+    with pytest.raises(ValueError):
+        catalog.create_table("table", schema=table_schema_simple)
+
+
+def test_create_table(table_schema_simple: Schema, hive_table: HiveTable):
+    catalog = HiveCatalog(HIVE_CATALOG_NAME, {}, url=HIVE_METASTORE_FAKE_URL)
+
+    catalog._client = MagicMock()
+    catalog._client.__enter__().create_table.return_value = hive_table
+    catalog._client.__enter__().get_table.return_value = hive_table
+
+    catalog.create_table(("default", "table"), schema=table_schema_simple)
+
+    catalog._client.__enter__().create_table.assert_called_with(
```

Comment:
Can you add `assert_called_with` for more of the test cases? I think we should validate what is passed to HMS wherever we can.

Reply:
Sure thing! Added them to all the happy path tests 👍🏻